app/process/handler/consensus-message/consensus-message.go

```diff
@@ -1,3 +1,110 @@
 package consensusmessage
 
-// TODO: Consensus message event handler
+import (
+	"encoding/hex"
+	"errors"
+	"fmt"
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/golang/protobuf/proto"
+	"github.com/limechain/hedera-eth-bridge-validator/app/domain/repositories"
+	"github.com/limechain/hedera-eth-bridge-validator/app/persistence/message"
+	"github.com/limechain/hedera-eth-bridge-validator/config"
+	validatorproto "github.com/limechain/hedera-eth-bridge-validator/proto"
+	log "github.com/sirupsen/logrus"
+	"strings"
+)
+
+type ConsensusMessageHandler struct {
+	repository     repositories.MessageRepository
+	validAddresses []string
+}
+
+func NewConsensusMessageHandler(r repositories.MessageRepository) *ConsensusMessageHandler {
+	return &ConsensusMessageHandler{
+		repository:     r,
+		validAddresses: config.LoadConfig().Hedera.Handler.ConsensusMessage.Addresses,
+	}
+}
+
+func (cmh ConsensusMessageHandler) Handle(payload []byte) {
+	err := cmh.handlePayload(payload)
+	if err != nil {
+		log.Errorf("Error - could not handle payload: [%s]", err)
+		return
+	}
+}
+
+func (cmh ConsensusMessageHandler) handlePayload(payload []byte) error {
+	m := &validatorproto.TopicSignatureMessage{}
+	err := proto.Unmarshal(payload, m)
+	if err != nil {
+		return errors.New(fmt.Sprintf("Failed to unmarshal topic signature message. - [%s]", err))
+	}
+
+	ctm := &validatorproto.CryptoTransferMessage{
+		TransactionId: m.TransactionId,
+		EthAddress:    m.EthAddress,
+		Amount:        m.Amount,
+		Fee:           m.Fee,
+	}
+
+	log.Printf("[%s] signature", m.GetSignature())
+
+	decodedSig, err := hex.DecodeString(m.GetSignature())
+	if err != nil {
+		return errors.New(fmt.Sprintf("Failed to decode signature. - [%s]", err))
+	}
+
+	hash := crypto.Keccak256([]byte(fmt.Sprintf("%s-%s-%d-%s", ctm.TransactionId, ctm.EthAddress, ctm.Amount, ctm.Fee)))
+
+	log.Printf("[%s] hex hash", hex.EncodeToString(hash))
+
+	key, err := crypto.Ecrecover(hash, decodedSig)
+	if err != nil {
+		return errors.New(fmt.Sprintf("Failed to recover public key. - [%s]", err))
+	}
+
+	pubKey, err := crypto.UnmarshalPubkey(key)
+	if err != nil {
+		return errors.New(fmt.Sprintf("Failed to unmarshal public key. - [%s]", err))
+	}
+
+	address := crypto.PubkeyToAddress(*pubKey)
+
+	if !cmh.isValidAddress(address.String()) {
+		return errors.New(fmt.Sprintf("Address is not valid - [%s]", address.String()))
+	}
+
+	messages, err := cmh.repository.GetByTxIdAndSignature(m.TransactionId, m.Signature)
```

Comment:
I think that for our next logic -> processing the TX if all signatures are acquired by the soft elected leader, we should get all TXns with this ID. In such case it would be better to change the `GetByTxIdAndSignature` to `GetTransaction` imo
and checking for duplicated signatures in the code here

Reply:
I will change the name of the method, but the duplication check is already done in the other branch for PR2.